### PR TITLE
Make ride window prices allow text input

### PR DESCRIPTION
**Pull request by wolfreak99**
This allows entering text into the price spinners on rides, using the same fashion as the cheat windows money does. I've left notes on each PR explaining what is going on, as I feel this would be a better approach than what I usually do by attempting to recollect all my thoughts into a PR's message (I also just found that comments are a possibility in commits)

I considered creating new text descriptions for the boxes, but eventually I would like to modify "window_event_textinput" to also include the key that was just pressed, that way it can read whether or not `return` or `escape` is pressed (much like the chat does), that way text can be modified directly within field boxes. That will have to be done later in a seperate PR, however. 

Something I feel I could do with this though, but I would like someones opinion, is switch all of the money16/32's to only handling a money32 outside of the get/set functions, and then have the get/set functions handle the conversion from one value to another. This would reduce the need of all of the currently placed castings.

Nevertheless, here's this in action as of the time of the original posting:
http://i.imgur.com/HjdvBDB.gifv

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Not sure why it's taken so long for someone to review this. Apologies for the wait... I've taken the liberty to rebase your PR, as it had a conflict.

The changes work well if the price input is as expected, e.g. `1.25`. However, there are a few outstanding issues, as far as I can tell:

- [ ] If a comma is input as a decimal place, instead of a period (e.g. `1,25`) we end up with `20.00`.
- [ ] If the price is set to something invalid, e.g. `aaa`, the price is set to free. I'd rather it not be changed at all.

Perhaps outside of the scope of this PR, but a would-be-nice:
- [ ] The price in the park window is currently unaffected.

**Comment by Gymnasiast** [comment]
> If the price is set to something invalid, e.g. aaa, the price is set to free. I'd rather it not be changed at all.

To this end, `string_to_money()` could be modified to return `MONEY32_UNDEFINED` if no usable characters are left after sanitising.

**Comment by Gymnasiast** [comment]
> If the price is set to something invalid, e.g. aaa, the price is set to free. I'd rather it not be changed at all.

This has now been addressed in another PR.

I do want to think this PR over a bit. I'm not sure if `try_get_primary_price` is the best way to approach this.

**Comment by AaronVanGeffen** [comment]
This PR will address #5038.

**Comment by Gymnasiast** [comment]
Not all of them. Most notably, the Scenario Editor window where you set cash is not (yet) addressed (though it would benefit most from this).

**Comment by AaronVanGeffen** [comment]
Nor the Park and Finances window, as I mentioned earlier. Would be nice, though!

**Comment by wolfreak99** [comment]
What would work best? if I was to first revamp the ride window price to allow direct input into the field (at the risk of the code becoming a little more complicated), or if I was to stick with using the external textinput window to handle the money for the rest of the price windows and worry about direct input later?

**Comment by Gymnasiast** [comment]
> > "To this end, string_to_money() could be modified to return MONEY32_UNDEFINED if no usable characters are left after sanitising."

> If you want, I can fix that for you. That would actually solve a part of the problem I mentioned in the original pr reguarding the textinput event handling pressing the escape. I wasn't sure if you self-assigning this means you taken it over though.

I have tackled this in #7516.

> I can also go through and modify the other windows with money input as well, if you wish.

I'd like to decide on that later.

File: src/openrct2-ui/windows/Ride.cpp [hunk 1]
```diff
@@ -5997,9 +6013,24 @@ static void window_ride_income_mouseup(rct_window *w, rct_widgetindex widgetInde
     case WIDX_TAB_10:
         window_ride_set_page(w, widgetIndex - WIDX_TAB_1);
         break;
+    case WIDX_PRIMARY_PRICE:{
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Ride.cpp]: Needs formatting fixes.

**Comment by Gymnasiast** [comment]
> If the current price has no cents, e.g. 1.00, the initial money in the input window has no decimals (just 1).

I have now also addressed this.

One thing that I did notice is that the `money_to_string()` and `string_to_money()` functions do not take the conversion factor into account. That will need to be addressed before merging.

**Comment by Gymnasiast** [comment]
Ok, I have now addressed that last point as well. I have also tested it with both pounds and yen.

**Comment by AaronVanGeffen** [comment]
This seems to be working fine, both with pounds and yen. Don't forget to add a changelog entry before merging. :-)

**Pull request merged**